src/z3c/jbot/utility.py

```diff
@@ -24,7 +25,7 @@ def getRequest():
             try:
                 return site.request
             except AttributeError:
-                return site.REQUEST
+                return getattr(site, "REQUEST", Zope2.app().REQUEST)
```

Comment:
This code always calls `Zope2.app().REQUEST` but it only seems to be needed in rare cases. I'd suggest to refactor this line, so `Zope2.app().REQUEST` is only computed when actually needed.